src/Sylius/Component/Core/spec/Resolver/ShippableCountriesResolverSpec.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Component\Core\Resolver;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Resolver\ShippableCountriesResolver;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+class ShippableCountriesResolverSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(ShippableCountriesResolver::class);
+    }
+
+    function let(RepositoryInterface $countryRepository, ChannelContextInterface $channelContext)
+    {
+        $this->beConstructedWith($countryRepository, $channelContext);
+    }
+
+    function it_returns_the_channels_shipping_countries(
```

Comment:
I think we need at least three tests in this specification:
1. Providing shippable countries for channel passed as a function argument
2. Providing shippable countries for channel got from context, when no channel is passed
3. Throwing exception/providing empty array of countries if there are no shippable countries on channel configured
**or** depending if [this comment](https://github.com/Sylius/Sylius/pull/9142/files#diff-ee05006a0528230d4de04f39e9236fb3R59) would be applied
3. Providing all countries if there are no shippable countries on channel configured